### PR TITLE
Provide Content Manifest API endpoint

### PR DESCRIPTION
**Pull request by athos-ribeiro**
Register a new “GET /requests/REQUEST_ID/content-manifest” API
endpoint. All requests should serve at least an empty, valid, content
manifest.

Assembling content manifests should be implemented in a per package
manager manner.

Signed-off-by: Athos Ribeiro <athos@redhat.com>

### CODE REVIEW

File: cachito/web/static/api_v1.yaml [hunk 1]
```diff
@@ -277,6 +277,33 @@ paths:
...
+        description: The ID of the Cachito request to retrieve the content manifest for
+        schema:
+          type: integer
+      responses:
+        "200":
+          description: The content manifest of the Cachito request
+          content:
+            application/json:
+              schema:
+                $ref: "https://raw.githubusercontent.com/containerbuildsystem/atomic-reactor/f4abcfdaf8247a6b074f94fa84f3846f82d781c6/atomic_reactor/schemas/content_manifest.json#"
```
Comment (athos-ribeiro) [inline cachito/web/static/api_v1.yaml:296]: Not sure how to get this one to work. It seems it is not happy with `"type": ["object", "null"]` and it just shows null. The file is being loaded, since the schema title is being imported.

**Comment by athos-ribeiro** [comment]
@mprahl, @lcarva, @ben-alkov, PTAL

**Comment by athos-ribeiro** [comment]
Partially adapted https://github.com/release-engineering/cachito/pull/173 here. Let's keep the other one open as a draft until we finish pulling the idea in. I will close it then.

File: cachito/web/models.py [hunk 2]
```diff
@@ -62,6 +62,29 @@
 )
 
 
+class ContentManifest:
+    """A content manifest associated with the request."""
+
+    version = 1
+    json_schema_url = (
+        "https://raw.githubusercontent.com/containerbuildsystem/atomic-reactor/"
+        "f4abcfdaf8247a6b074f94fa84f3846f82d781c6/atomic_reactor/schemas/content_manifest.json"
```
Comment (ben-alkov) [inline cachito/web/models.py:71]: Using a specific commit hash seems brittle...

Comment (mprahl) [reply cachito/web/models.py:71]: Could you use a Git tag instead of a commit hash?

File: cachito/web/models.py [hunk 3]
```diff
@@ -62,6 +62,29 @@
 )
 
 
+class ContentManifest:
```
Comment (mprahl) [inline cachito/web/models.py:65]: Could you move this to a new file? `models.py` is already rather large and this change is not really related to a database schema.

File: cachito/web/static/api_v1.yaml (hunk 1 above)
Comment (mprahl) [reply cachito/web/static/api_v1.yaml:296]: I'm not sure JSON schema is supported as OpenAPI is an extended subset of JSON schema:
https://swagger.io/docs/specification/data-models/keywords/

File: cachito/web/models.py (hunk 2 above)
Comment (athos-ribeiro) [reply cachito/web/models.py:71]: The reason I used the commit in the first place was due to the floating nature of tags. I can move to a tag, knowing we do control the tags in atomic-reactor and they will not change.

File: cachito/web/static/api_v1.yaml (hunk 1 above)
Comment (athos-ribeiro) [reply cachito/web/static/api_v1.yaml:296]: Any suggestions to avoid duplicating the schema here?

File: cachito/web/models.py (hunk 2 above)
Comment (athos-ribeiro) [reply cachito/web/models.py:71]: Actually, we do not have an atomic-reactor release with it, so we would need to wait for the next release cycle to change this. Any other suggestions?

File: cachito/web/static/api_v1.yaml (hunk 1 above)
Comment (athos-ribeiro) [reply cachito/web/static/api_v1.yaml]: I ended up partially pulling in the schema

File: cachito/web/models.py (hunk 2 above)
Comment (mprahl) [reply cachito/web/models.py:71]: @athos-ribeiro that's fine. Let's just update it once it's released or you can set it now knowing the URL is invalid until the release.

File: cachito/web/static/api_v1.yaml [hunk 4]
```diff
@@ -277,6 +277,75 @@ paths:
...
+        schema:
+          type: integer
+      responses:
+        "200":
+          description: The content manifest of the Cachito request
+          content:
+            application/json:
+              schema:
+                type: object
+                nullable: true
```
Comment (mprahl) [inline cachito/web/static/api_v1.yaml:297]: Is this actually true?

File: cachito/web/static/api_v1.yaml [hunk 5]
```diff
@@ -277,6 +277,75 @@ paths:
...
+                properties:
+                  metadata:
+                    type: object
+                    properties:
+                      icm_version:
+                        type: integer
+                        example: 1
+                      icm_spec:
+                        type: string
+                        example: "https://json-schema.icm.example.com"
```
Comment (mprahl) [inline cachito/web/static/api_v1.yaml:307]: A real example would be nice here so people can go to the schema directly from the Cachito API documentation.

File: cachito/web/static/api_v1.yaml [hunk 6]
```diff
@@ -277,6 +277,75 @@ paths:
...
+                      image_layer_index:
+                        type: integer
+                        example: 1
```
Comment (mprahl) [inline cachito/web/static/api_v1.yaml:310]: Will Cachito ever return an integer other than `-1` since it has no context of the image that the request is used for?

File: cachito/web/static/api_v1.yaml [hunk 7]
```diff
@@ -277,6 +277,75 @@ paths:
...
+                        purl:
+                          $ref: "#/components/schemas/Purl"
+                        dependencies:
+                          type: array
+                          items:
+                            type: object
+                            properties:
+                              purl:
+                                $ref: "#/components/schemas/Purl"
+                        sources:
```
Comment (mprahl) [inline cachito/web/static/api_v1.yaml:325]: Will Cachito ever set this? If not, I think this should be removed.

File: cachito/web/static/api_v1.yaml [hunk 8]
```diff
@@ -277,6 +277,75 @@ paths:
...
+                          type: array
+                          items:
+                            type: object
+                            properties:
+                              purl:
+                                $ref: "#/components/schemas/Purl"
+                  content_sets:
```
Comment (mprahl) [inline cachito/web/static/api_v1.yaml:332]: Will Cachito ever set this? If not, I think this should be removed.

**Comment by mprahl** [comment]
@athos-ribeiro this looks good. I left a few comments. Once addressed, please squash the commits and then it should be good to merge.

File: cachito/web/static/api_v1.yaml (hunk 4 above)
Comment (athos-ribeiro) [reply cachito/web/static/api_v1.yaml:297]: According to the original schema, it is. We could remove it here since we are not returning null though, since this is just for documenting what cachito is returning.

File: cachito/web/static/api_v1.yaml (hunk 7 above)
Comment (athos-ribeiro) [reply cachito/web/static/api_v1.yaml:325]: Yes, the idea is that cachito itself will return this one filled for gomod with the module level dependencies, which were the sources used in build time

File: cachito/web/static/api_v1.yaml (hunk 8 above)
Comment (athos-ribeiro) [reply cachito/web/static/api_v1.yaml:332]: Not really, I will remove this one

File: cachito/web/static/api_v1.yaml (hunk 4 above)
Comment (athos-ribeiro) [reply cachito/web/static/api_v1.yaml:297]: Removed

File: cachito/web/static/api_v1.yaml (hunk 5 above)
Comment (athos-ribeiro) [reply cachito/web/static/api_v1.yaml:307]: Changed

File: cachito/web/static/api_v1.yaml (hunk 6 above)
Comment (athos-ribeiro) [reply cachito/web/static/api_v1.yaml:310]: Changed to `-1`

File: cachito/web/static/api_v1.yaml (hunk 8 above)
Comment (athos-ribeiro) [reply cachito/web/static/api_v1.yaml:332]: removed

**Comment by athos-ribeiro** [comment]
Comments addressed; commits squashed

File: cachito/web/static/api_v1.yaml (hunk 7 above)
Comment (mprahl) [reply cachito/web/static/api_v1.yaml:326]: I reread the design document and now remember what sources is for.

File: cachito/web/static/api_v1.yaml [hunk 9]
```diff
@@ -277,6 +277,69 @@ paths:
...
+                    type: object
+                    properties:
+                      icm_version:
+                        type: integer
+                        example: 1
+                      icm_spec:
+                        type: string
+                        example: "https://raw.githubusercontent.com/containerbuildsystem/\
+                                  atomic-reactor/f4abcfdaf8247a6b074f94fa84f3846f82d781c6/\
+                                  atomic_reactor/schemas/content_manifest.json"
```
Comment (mprahl) [inline cachito/web/static/api_v1.yaml:308]: I think this might be the preferred syntax for long strings in YAML:
```suggestion
                        example: >-
                          https://raw.githubusercontent.com/containerbuildsystem/
                          atomic-reactor/f4abcfdaf8247a6b074f94fa84f3846f82d781c6/
                          atomic_reactor/schemas/content_manifest.json"
```

Comment (athos-ribeiro) [reply cachito/web/static/api_v1.yaml:308]: Wouldn't this add spaces after the slashes though?

Comment (mprahl) [reply cachito/web/static/api_v1.yaml:308]: You're right
